employees/views.py

```diff
@@ -158,34 +166,48 @@ def redirect_to_current_month(self) -> HttpResponseRedirectBase:
     ),
     name="dispatch",
 )
-class ReportList(UserIsManagerOfCurrentReportProjectMixin, UserIsAuthorOfCurrentReportMixin, APIView):
+class ReportList(
```

Comment:
Same here about getting along with https://github.com/Code-Poets/sheetstorm/pull/274

Now when I see how this view grew, I think we should separate creating from displaying list. But this is just my opinion, which is driven by the fears it will be harder to maintain the current version. But it is rather a business decision, so please just take my opinion into consideration.

Reply:
I also think this might be a good idea. It was convenient at first, but now it's getting harder to maintain and develop. This way, we could also connect together all the list-related views with single parent class, as @dybi suggested. This way, I could also just move all the functionality of **MonthSegregationMixin** to such view (since we still need to overload `get()` and `post()`  in each view to get it to work 100%).

Reply:
I personally think that firstly we should merge everything what is necessary and in next step we start to change code without changing functionalities. There are 3 PRs left to merge to start refactoring code as @rwrzesien suggested at start. So now I will merge it and after we merge all Maciek's pull requests we should all have little talk how we want to do this. @rwrzesien @dybi @maciejSamerdak @MartynaAnnaGottschling @Szymiks @Karrp

Reply:
> I personally think that firstly we should merge everything what is necessary and in next step we start to change code without changing functionalities.

Off course, this is what I have meant too.

> we should all have little talk how we want to do this

Please add me to a meeting, if I will be able I would like to participate.